src/blob_file_builder.cc

```diff
@@ -33,19 +33,35 @@ BlobFileBuilder::BlobFileBuilder(const TitanDBOptions& db_options,
     return;
 #endif
   }
+  block_size_ = cf_options.enable_punch_hole_gc ? cf_options.block_size : 0;
   WriteHeader();
 }
 
+void BlobFileBuilder::FillBlockWithPad() {
+  if (block_size_ == 0) {
+    return;
+  }
+  size_t pad_size = block_size_ - (file_->GetFileSize() % block_size_);
+  if (pad_size == block_size_) {
+    return;
+  }
+  std::string pad(pad_size, 0);
```

Comment:
Can we use a static constant 4K length 0 array, and use ref as the input of append to eliminate the dynamic construction & destruction of string?

Reply:
Done